File: 7_kyu/Even_numbers_in_an_array.kt
fun evenNumbers(array: List<Int>, number: Int): List<Int> {
    val acc = array.foldRight( ArrayList<Int>(number) ) {
        it, acc -> if(acc.size < number && it % 2 == 0) acc.add(it); acc
    }
    return acc.reversed()
}

File: 7_kyu/Even_numbers_in_an_array.py
def even_numbers(arr: list[int],n :int) -> list[int]:
    """
    Returns the last n even numbers in list arr.
    """
    result = []
    for x in reversed(arr):
        if x%2 == 0:
            result.insert(0,x)
            if len(result) == n:
                return result
    return result